src/CachePeer.cc

```diff
@@ -123,6 +126,207 @@ CachePeer::connectTimeout() const
+        os << " login=" << login;
+
+    if (mcast.ttl > 0)
+        os << " ttl=" << mcast.ttl;
+
+    if (connect_timeout_raw > 0)
+        os << " connect-timeout=" << connect_timeout_raw;
+
+    if (connect_fail_limit != PEER_TCP_MAGIC_COUNT)
+        os << " connect-fail-limit=" << connect_fail_limit;
+
+#if USE_CACHE_DIGESTS
+    if (digest_url)
+        os << " digest-url=" << digest_url;
+#endif
+
+    if (options.allow_miss)
+        os << " allow-miss";
+
+    if (options.no_tproxy)
+        os << " no-tproxy";
+
+    if (max_conn > 0)
+        os << " max-conn=" << max_conn;
+
+    if (standby.limit > 0)
+        os << " standby=" << standby.limit;
+
+    if (options.originserver)
+        os << " originserver";
+
+    if (domain)
+        os << " forceddomain=" << domain;
+
+    if (connection_auth == 0)
+        os << " connection-auth=off";
+    else if (connection_auth == 1)
+        os << " connection-auth=on";
+    else if (connection_auth == 2)
+        os << " connection-auth=auto";
+
+    secure.dumpCfg(os, "tls-");
+}
+
+void
+CachePeer::reportStatistics (std::ostream& os)
+{
+    os <<
+       "Name       : " << name << '\n' <<
+       "Type       : " << typeString() << '\n' <<
+       "Addresses  : " << n_addresses << '\n' <<
+       "Host       : " << host << '/' << http_port << '/' <<
+       icp.port << '\n';
+    os << "Flags      :";
+    dumpOptions(os);
+    os << '\n';
+
+    char ntoabuf[MAX_IPSTRLEN];
+    for (int i = 0; i < n_addresses; ++i)
+        os << "Address[" << i << "] : " <<
+           addresses[i].toStr(ntoabuf, MAX_IPSTRLEN) << '\n';
+
+    os << "Status     : " << (neighborUp(this) ? "Up" : "Down") << '\n' <<
+       "Fetches    : " << stats.fetches << '\n' <<
+       "Open conns : " << stats.conn_open << '\n' <<
+       "Average RTT: " << stats.rtt << " msec\n";
+
+    if (stats.last_query > 0)
+        os << "Last query : " <<
+           (squid_curtime - stats.last_query) << " seconds ago\n";
+
+    if (stats.last_reply > 0)
+        os << "Last reply : " <<
+           (squid_curtime - stats.last_reply) << " seconds ago\n";
+
+    os << "Pings sent : "  << stats.pings_sent << '\n' <<
+       "Pings acked: " << stats.pings_acked << " " <<
+       Math::intPercent(stats.pings_acked, stats.pings_sent) << "%\n";
+
+    os << "Ignored    : " << stats.ignored_replies << " " <<
+       Math::intPercent(stats.ignored_replies, stats.pings_acked) << "%\n";
+
+    os << "Histogram of pings acked:\n";
+
+#if USE_HTCP
+    if (options.htcp) {
+        os << "\tMisses\t" << htcp.counts[0] << " " <<
+           Math::intPercent(htcp.counts[0], stats.pings_acked) << "%\n" <<
+           "\tHits\t" << htcp.counts[1] << " " <<
+           Math::intPercent(htcp.counts[1], stats.pings_acked) << "%\n";
+    } else {
+#endif
+        for (auto op : WholeEnum<icp_opcode>()) {
+            if (icp.counts[op] == 0)
+                continue;
+
+            os << "    " << std::setw(12) << std::setprecision(12) <<
```

Comment:
The official code does not use 12-digit precision. Moreover, this code does not print floating point numbers so setprecision() cannot be useful here.

The best way to address this and other related (but not yet spelled out) concerns is to add an AsPercent class to IoManip.h. The constructor should take part and whole arguments...

Reply:
AsPercent seems a bit overkill for now, I'll just avoid setting the precision - I suspect
it comes out of printf() formatting, which we are doing away with anyway

Reply:
removed the precision, if we rely on yaml instead of spacing it's completely unnecessary anyway